satpy/readers/utils.py

```diff
@@ -210,57 +230,84 @@ def unzip_file(filename, prefix=None):
         Temporary filename path for decompressed file or None.
 
     """
-    if os.fspath(filename).endswith('bz2'):
-        fdn, tmpfilepath = tempfile.mkstemp(prefix=prefix,
-                                            dir=config["tmp_dir"])
-        LOGGER.info("Using temp file for BZ2 decompression: %s", tmpfilepath)
-        # try pbzip2
-        pbzip = which('pbzip2')
-        # Run external pbzip2
-        if pbzip is not None:
-            n_thr = os.environ.get('OMP_NUM_THREADS')
-            if n_thr:
-                runner = [pbzip,
-                          '-dc',
-                          '-p'+str(n_thr),
-                          filename]
-            else:
-                runner = [pbzip,
-                          '-dc',
-                          filename]
-            p = Popen(runner, stdout=PIPE, stderr=PIPE)  # nosec
-            stdout = BytesIO(p.communicate()[0])
-            status = p.returncode
-            if status != 0:
-                raise IOError("pbzip2 error '%s', failed, status=%d"
-                              % (filename, status))
-            with closing(os.fdopen(fdn, 'wb')) as ofpt:
-                try:
-                    stdout.seek(0)
-                    shutil.copyfileobj(stdout, ofpt)
-                except IOError:
-                    import traceback
-                    traceback.print_exc()
-                    LOGGER.info("Failed to read bzipped file %s",
-                                str(filename))
-                    os.remove(tmpfilepath)
-                    raise
-            return tmpfilepath
-
-        # Otherwise, fall back to the original method
-        bz2file = bz2.BZ2File(filename)
+    if not os.fspath(filename).endswith('bz2'):
+        return None
+
+    fdn, tmpfilepath = tempfile.mkstemp(prefix=prefix,
+                                        dir=config["tmp_dir"])
+    LOGGER.info("Using temp file for BZ2 decompression: %s", tmpfilepath)
+    # try pbzip2
+    pbzip = which('pbzip2')
+    # Run external pbzip2
+    if pbzip is not None:
```

Comment:
Is it possible to put this entire `pbzip/which` and `if` statement in a separate helper function like `_decompress_with_pbzip_if_available`? If it returns the tmpfilepath then we know it was successful and we can return `tmpfilepath`. If it is `None` then we know `pbzip` wasn't available and we should continue with the `bz2` module decompression.